### PR TITLE
[MAINTENANCE] mypy 1.11.2

### PR DESCRIPTION
**Pull request by Kilo59**
https://mypy-lang.blogspot.com/2024/07/mypy-111-released.html

Lots of new typing errors, mostly relating to the need to explicitly add `@override` in cases where a Python built-in is being subclassed.

This PR contains no runtime changes.

Used `mypy-silent` to auto-add type ignore comments.
Thank you @anthonyburdi for sharing this tool.
https://github.com/whtsky/mypy-silent

### CODE REVIEW

File: contrib/cli/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Click>=7.1.2         # CLI tooling
 cookiecutter==2.1.1  # Project templating
-mypy==1.10.1            # Type checker
+mypy==1.11            # Type checker
```
Comment (Kilo59) [inline contrib/cli/requirements.txt:3]:
```suggestion
mypy==1.11.2            # Type checker
```

File: reqs/requirements-dev-contrib.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 adr-tools-python==1.0.3
 invoke>=2.0.0
-mypy==1.10
+mypy==1.11
```
Comment (Kilo59) [inline reqs/requirements-dev-contrib.txt:3]:
```suggestion
mypy==1.11.2
```

File: great_expectations/compatibility/databricks.py [hunk 3]
```diff
@@ -5,6 +5,6 @@
 )
 
 try:
-    from databricks import connect  # type: ignore[import-not-found]
+    from databricks import connect  # type: ignore[import-not-found]  # type: ignore[import-untyped]
```
Comment (Kilo59) [inline great_expectations/compatibility/databricks.py:8]:
```suggestion
    from databricks import connect  # type: ignore[import-not-found,import-untyped]
```

**Comment by Kilo59** [comment]
I started to add `@override` to all of the violations but I got lazy.

File: great_expectations/compatibility/typing_extensions.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
     from typing_extensions import Annotated
 
 try:
-    from typing_extensions import override
+    from great_expectations.compatibility.typing_extensions import override
```
Comment (Kilo59) [inline great_expectations/compatibility/typing_extensions.py:13]:
```suggestion
    from typing_extensions import override
```

File: great_expectations/core/expectation_suite.py [hunk 5]
```diff
@@ -282,7 +282,7 @@ def expectation_configurations(self, value):
             "Please use ExpectationSuite.expectations instead."
         )
 
-    def __eq__(self, other):
+    def __eq__(self, other):  # type: ignore[explicit-override] # FIXME
```
Comment (tyler-hoffman) [inline great_expectations/core/expectation_suite.py:285]: any reason why some of these were given override decorators and some given type ignores?

Comment (Kilo59) [reply great_expectations/core/expectation_suite.py:285]: No, other than I came down with a sudden case of lazyness.

**Review by tyler-hoffman — APPROVED** [review]
Thanks as always!

File: great_expectations/core/expectation_suite.py (hunk 5 above)
Comment (tyler-hoffman) [reply great_expectations/core/expectation_suite.py:285]: 🙈

**Pull request merged**